File: code_snippets/api-guide/data/create_get_update_delete/java/delete_input_by_id.java
package com.clarifai.example;

import com.clarifai.channel.ClarifaiChannel;
import com.clarifai.credentials.ClarifaiCallCredentials;
import com.clarifai.grpc.api.*;
import com.clarifai.grpc.api.status.*;

public class ClarifaiExample {

    ///////////////////////////////////////////////////////////////////////////////////
    // In this section, we set the user authentication, app ID, and the ID of the
    // input we want to delete. Change these strings to run your own example
    ///////////////////////////////////////////////////////////////////////////////////

    static final String USER_ID =  "YOUR_USER_ID_HERE";
    //Your PAT (Personal Access Token) can be found in the portal under Authentication
    static final String PAT = "YOUR_PAT_HERE";
    static final String APP_ID = "YOUR_APP_ID_HERE";
    // Change this based on the input you want to delete
    static final String INPUT_ID = "ff79664eefe94db1878f51931f9d6fd9";
       
    ///////////////////////////////////////////////////////////////////////////////////
    // YOU DO NOT NEED TO CHANGE ANYTHING BELOW THIS LINE TO RUN THIS EXAMPLE
    ///////////////////////////////////////////////////////////////////////////////////

    public static void main(String[] args) {

        V2Grpc.V2BlockingStub stub = V2Grpc.newBlockingStub(ClarifaiChannel.INSTANCE.getGrpcChannel())
            .withCallCredentials(new ClarifaiCallCredentials(PAT));            

        BaseResponse deleteInputResponse = stub.deleteInput(
            DeleteInputRequest.newBuilder()
               .setUserAppId(UserAppIDSet.newBuilder().setUserId(USER_ID).setAppId(APP_ID))
                .setInputId(INPUT_ID)
                .build()
        );

        if (deleteInputResponse.getStatus().getCode() != StatusCode.SUCCESS) {
            throw new RuntimeException("Delete input failed, status: " + deleteInputResponse.getStatus());
        }  
                       
    }  
}

File: code_snippets/api-guide/data/create_get_update_delete/py/delete_input_by_id.py
##############################################################################
# In this section, we set the user authentication, app ID, and the ID of the 
# input we want to delete. Change these strings to run your own example.
##############################################################################

USER_ID = 'YOUR_USER_ID_HERE'
# Your PAT (Personal Access Token) can be found in the Account's Security section
PAT = 'YOUR_PAT_HERE'
APP_ID = 'YOUR_APP_ID_HERE'
# Change this based on the input you want to delete
INPUT_ID = '2e9c4a86555d40ffb47c7b045d7e3048'

##########################################################################
# YOU DO NOT NEED TO CHANGE ANYTHING BELOW THIS LINE TO RUN THIS EXAMPLE
##########################################################################

from clarifai_grpc.channel.clarifai_channel import ClarifaiChannel
from clarifai_grpc.grpc.api import resources_pb2, service_pb2, service_pb2_grpc
from clarifai_grpc.grpc.api.status import status_code_pb2

channel = ClarifaiChannel.get_grpc_channel()
stub = service_pb2_grpc.V2Stub(channel)

metadata = (('authorization', 'Key ' + PAT),)

userDataObject = resources_pb2.UserAppIDSet(user_id=USER_ID, app_id=APP_ID)

delete_input_response = stub.DeleteInput(
    service_pb2.DeleteInputRequest(
    user_app_id=userDataObject, 
    input_id=INPUT_ID
    ),
    metadata=metadata
)

if delete_input_response.status.code != status_code_pb2.SUCCESS:
    print(delete_input_response.status)
    raise Exception("Delete input failed, status: " + delete_input_response.status.description)
